modules/encrypt/classes/Kohana/Encrypt.php

```diff
@@ -0,0 +1,147 @@
+<?php defined('SYSPATH') OR die('No direct script access.');
+
+class Kohana_Encrypt {
+
+	/**
+	 * @var  string  default instance name
+	 */
+	public static $default = 'default';
+
+	/**
+	 * @var  array  Encrypt class instances
+	 */
+	public static $instances = array();
+
+	/**
+	 * @var  engine  Encryption engine
+	 */
+	public $_engine = NULL;
+
+	/**
+	 * Returns a singleton instance of Encrypt. An encryption key must be
+	 * provided in your "encrypt" configuration file.
+	 *
+	 *     $encrypt = Encrypt::instance();
+	 *
+	 * @param   string  $name   configuration group name
+	 * @return  Encrypt
+	 */
+	public static function instance($name = NULL, array $config = NULL)
+	{
+		if ($name === NULL)
+		{
+			// Use the default instance name
+			$name = Encrypt::$default;
+		}
+
+		if ( ! isset(Encrypt::$instances[$name]))
+		{
+			if ($config === NULL)
+			{
+				// Load the configuration data
+				$config = Kohana::$config->load('encrypt')->$name;
+			}
+
+			if ( ! isset($config['key']))
+			{
+				// No default encryption key is provided!
+				throw new Kohana_Exception('No encryption key is defined in the encryption configuration group: :group',
+					array(':group' => $name));
+			}
+
+			if ( ! isset($config['mode']))
+			{
+				// Add the default mode
+				$config['mode'] = MCRYPT_MODE_NOFB;
+			}
+
+			if ( ! isset($config['cipher']))
+			{
+				// Add the default cipher
+				$config['cipher'] = MCRYPT_RIJNDAEL_128;
```

Comment:
shouldn't the cipher type be changed to aes-256-ctr (or aes-256-cbc) for that matter?